x-pack/plugin/core/src/main/resources/data-streams-mappings.json

```diff
@@ -0,0 +1,73 @@
+{
+  "template": {
+    "mappings": {
+      "dynamic_templates": [
+        {
+          "match_ip": {
+            "match_mapping_type": "string",
+            "match": "ip",
+            "mapping": {
+              "type": "ip"
+            }
+          }
+        },
+        {
+          "strings_as_keyword": {
+            "mapping": {
+              "ignore_above": 1024,
+              "type": "keyword"
+            },
+            "match_mapping_type": "string"
+          }
+        }
+      ],
+      "date_detection": false,
+      "properties": {
+        "@timestamp": {
+          "type": "date"
+        },
+        "data_stream": {
+          "properties": {
+            "type": {
+              "type": "constant_keyword",
+              "value": "synthetics"
```

Comment:
This template says it's for general mapping conventions, but this uses `synthetics` which means this can't be generalized for other types? I think we might need to remove this line since this is used by the logs and metrics templates which previously specified the value as a different value